src/dash-playlist-loader.js

```diff
@@ -472,11 +651,51 @@ export default class DashPlaylistLoader extends EventTarget {
 
       this.masterXml_ = req.responseText;
 
-      const newMaster = this.parseMasterXml();
-      const updatedMaster = updateMaster(this.master, newMaster);
+      // This will filter out updated sidx info from the mapping
+      this.sidxMapping_ = filterSidxMapping(
+        this.masterXml_,
+        this.srcUrl,
+        this.clientOffset_,
+        this.sidxMapping_
+      );
+
+      const master = this.parseMasterXml();
+      const updatedMaster = updateMaster(this.master, master);
 
       if (updatedMaster) {
-        this.master = updatedMaster;
+        const sidxKey = generateSidxKey(this.media().sidx);
+
+        // the sidx was updated, so the previous mapping was removed
+        if (!this.sidxMapping_[sidxKey]) {
+          const playlist = this.media();
+
+          this.request = requestSidx_(
+            playlist.sidx,
+            playlist,
+            this.hls_.xhr,
+            { handleManifestRedirects: this.handleManifestRedirects },
+            this.handleSidxResponse_(playlist, master, this.state, (newMaster, sidx) => {
+              if (!newMaster || !sidx) {
```

Comment:
would it make more sense for this to be `getSidxFromResponse_` since it isn't actually doing all of the handling.

Reply:
maybe, I based the naming off of segmentLoader. Looks like that has `segmentRequestFinished_` and `processSegmentResponse_` which might be better ones to follow for naming

Reply:
went with sidxRequestFinished_ to match segment loader.